From b7e5e91642bb8f6d7439dee991e7553840373dff Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@nxp.com>
Date: Thu, 15 Feb 2018 15:30:51 -0600
Subject: [PATCH 3366/5242] MLK-17597-1: i.MX8MQ workaround affecting other
 SoCs

commit  0a693525ea4d17f64560a082c245dbe740db4354 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix iMX8MQ workaround to be specific to that
machine.

Signed-off-by: Nitin Garg <nitin.garg@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/irqchip/irq-gic-v3.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/irqchip/irq-gic-v3.c b/drivers/irqchip/irq-gic-v3.c
index c676fca..3ec9eb9 100644
--- a/drivers/irqchip/irq-gic-v3.c
+++ b/drivers/irqchip/irq-gic-v3.c
@@ -1323,8 +1323,11 @@ static int __init gic_of_init(struct device_node *node, struct device_node *pare
 		rdist_regs[i].phys_base = res.start;
 	}
 
-	/* sw workaround for IPI can't wakeup CORE ERRATA(ERR011171) on i.MX8MQ */
-	iomuxc_gpr_base = of_iomap(node, 2);
+	if (of_machine_is_compatible("fsl,imx8mq")) {
+		/* sw workaround for IPI can't wakeup CORE
+		   ERRATA(ERR011171) on i.MX8MQ */
+		iomuxc_gpr_base = of_iomap(node, 2);
+	}
 
 	if (of_property_read_u64(node, "redistributor-stride", &redist_stride))
 		redist_stride = 0;
-- 
1.7.9.5

